backend/controller/evaljob_controller.go

```diff
@@ -0,0 +1,568 @@
+		}
+	}
+
+	err := r.Status().Update(ctx, job)
+	if err != nil {
+		log.Error(err, "failed to update status for cancellation")
+	}
+	r.Recorder.Event(job, "Normal", "Cancelled",
+		fmt.Sprintf("Tthe EvalJob %s in namespace %s has cancelled and changed its state to Complete",
+			job.Name,
+			job.Namespace))
+	return ctrl.Result{}, err
+}
+
+func (r *EvalJobReconciler) createPod(job *lmevalservicev1beta1.EvalJob) *corev1.Pod {
+	var allowPrivilegeEscalation = false
+	var runAsNonRootUser = true
+	var ownerRefController = true
+	var runAsUser int64 = 1001030000
+
+	// Then compose the Pod CR
+	pod := corev1.Pod{
+		TypeMeta: v1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: v1.ObjectMeta{
+			Name:      job.Name,
+			Namespace: job.Namespace,
+			OwnerReferences: []v1.OwnerReference{
+				{
+					APIVersion: job.APIVersion,
+					Kind:       job.Kind,
+					Name:       job.Name,
+					Controller: &ownerRefController,
+					UID:        job.UID,
+				},
+			},
+			Labels: map[string]string{
+				"app.kubernetes.io/name": "fms-lm-eval-service",
+			},
+		},
+		Spec: corev1.PodSpec{
+			InitContainers: []corev1.Container{
+				{
+					Name:            "driver",
+					Image:           r.options.DriverImage,
+					ImagePullPolicy: corev1.PullAlways,
+					Command:         []string{DriverPath, "--copy", DestDriverPath},
+					SecurityContext: &corev1.SecurityContext{
+						AllowPrivilegeEscalation: &allowPrivilegeEscalation,
+						RunAsUser:                &runAsUser,
+						Capabilities: &corev1.Capabilities{
+							Drop: []corev1.Capability{
+								"ALL",
+							},
+						},
+					},
+					VolumeMounts: []corev1.VolumeMount{
+						{
+							Name:      "shared",
+							MountPath: "/opt/app-root/src/bin",
+						},
+					},
+				},
+			},
+			Containers: []corev1.Container{
+				{
+					Name:            "main",
+					Image:           r.options.PodImage,
+					ImagePullPolicy: corev1.PullAlways,
+					Env: []corev1.EnvVar{
+						{
+							Name: "GENAI_KEY",
+							ValueFrom: &corev1.EnvVarSource{
+								SecretKeyRef: &corev1.SecretKeySelector{
+									Key: "key",
+									LocalObjectReference: corev1.LocalObjectReference{
+										Name: "genai-key",
+									},
+								},
+							},
+						},
+					},
+					Command: generateCmd(job),
+					Args:    generateArgs(job),
+					SecurityContext: &corev1.SecurityContext{
+						AllowPrivilegeEscalation: &allowPrivilegeEscalation,
+						RunAsUser:                &runAsUser,
+						Capabilities: &corev1.Capabilities{
+							Drop: []corev1.Capability{
+								"ALL",
+							},
+						},
+					},
+					VolumeMounts: []corev1.VolumeMount{
+						{
+							Name:      "shared",
+							MountPath: "/opt/app-root/src/bin",
+						},
+					},
+				},
+			},
+			SecurityContext: &corev1.PodSecurityContext{
+				RunAsNonRoot: &runAsNonRootUser,
+				SeccompProfile: &corev1.SeccompProfile{
+					Type: corev1.SeccompProfileTypeRuntimeDefault,
```

Comment:
Somewhere in the back of my mind, this is different depending on the version of kubernetes/openshift you're running. We probably don't need to go back in time too far though?

Reply:
I lost track too. but most of the recent k8s clusters need it.